platform/security/pdp/security-pdp-xacmlrealm/src/main/java/ddf/security/pdp/xacml/processor/BalanaPdp.java

```diff
@@ -363,26 +352,34 @@ private String marshal(RequestType xacmlRequestType) throws PdpException {
     /**
      * Unmarshalls the XACML response.
      *
-     * @param xacmlResponse
-     *            The XACML response with all namespaces and namespace prefixes added.
+     * @param xacmlResponse The XACML response with all namespaces and namespace prefixes added.
      * @return The XACML response.
      * @throws PdpException
      */
+    @SuppressWarnings("unchecked")
     private ResponseType unmarshal(DOMResult xacmlResponse) throws PdpException {
+        List<String> ctxPath = ImmutableList.of(ResponseType.class.getPackage()
+                .getName());
+
+        if (null == parser) {
```

Comment:
If you execute this test first, you save yourself the expense of creating an object above.